### PR TITLE
[WIP] sklearn API for Gensim models

### PR DESCRIPTION
**Pull request by chinmayapancholi13**
This PR is for creating scikit-learn API for the following Gensim models : 
- [x] Doc2Vec Model
- [x] Text to Bag-of-words Model
- [x] Term Frequency-Inverse Document Frequency Model
- [x] Hierarchical Dirichlet Process Model
- [x] Phrases
- [ ] Normalization Model
- [ ] LogEntropy Model
- [ ] Dynamic Topic Model
- [ ] Topic Coherence Model

### CODE REVIEW

**Comment by tmylk** [comment]
Are you sure there is no memory duplication for doc2vec numpy arrays? can you please run it through a memory profiler.

File: gensim/sklearn_integration/sklearn_wrapper_gensim_text2bow.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        return self
+
+    def transform(self, docs):
+        """
+        Return the BOW format for the input documents.
+        """
+        if self.gensim_model is None:
+            raise NotFittedError("This model has not been fitted yet. Call 'fit' with appropriate arguments before using this method.")
+
+        # The input as array of array
```
Comment (tmylk) [inline gensim/sklearn_integration/sklearn_wrapper_gensim_text2bow.py]: Please call them python lists

**Comment by tmylk** [comment]
Please add ipynb examples in a separate notebook for now. We will copy-paste the notebooks together when these models.

File: gensim/test/test_sklearn_integration.py [hunk 2]
```diff
@@ -537,5 +571,302 @@ def testPipeline(self):
...
+        self.assertEqual(matrix.shape[0], 3)
+        self.assertEqual(matrix.shape[1], self.model.size)
+
+        # tranform one document
+        doc = w2v_texts[0]
+        matrix = self.model.transform(doc)
+        self.assertEqual(matrix.shape[0], 1)
+        self.assertEqual(matrix.shape[1], self.model.size)
+
+    def testSetGetParams(self):
```
Comment (menshikh-iv) [inline gensim/test/test_sklearn_integration.py]: please add checking with the original model too for each "getset" test (same as previous PR)

Comment (chinmayapancholi13) [inline gensim/test/test_sklearn_integration.py]: Done.

**Comment by menshikh-iv** [comment]
Grate @chinmayapancholi13 :100: 
You gave a new interface for gensim:+1:
